### PR TITLE
Remove password from stringified outputs

### PR DESCRIPTION
**Pull request by charmander**

### CODE REVIEW

File: packages/pg/lib/client.js [hunk 1]
```diff
@@ -33,12 +33,11 @@ var Client = function (config) {
 
   // "hiding" the password so it doesn't show up in stack traces
   // or if the client is console.logged
-  const password = this.connectionParameters.password
   Object.defineProperty(this, 'password', {
+    configurable: true,
```
Comment (brianc) [inline packages/pg/lib/client.js]: i didn't really understand the advantage or diff between configurable or not 🤷‍♂

Comment (charmander) [reply packages/pg/lib/client.js]: Configurable is pretty much always the right option. It lets people monkeypatch, hook, freeze, and do all the other JavaScripty stuff when they want to, and it’s the default for most things built into the language.

Except `Object.defineProperty`. 🙃

**Comment by charmander** [comment]
I just realized that 410b89b22bd72cbdb71db5de8c990d16a2cf6c04 fixes the problem for `Pool` when used from pg-pool, but not when used from pg (because its `BoundPool` extends the pool’s options with an `Object.assign`). Should the same fix go in there to support `new pg.Pool(otherPool.options)`, or does that seem like too much of a hack to avoid swapping out `Object.assign` with something that doesn’t care about whether properties are inherited or enumerable? @brianc

**Comment by brianc** [comment]
I'm not following what you're saying exactly? If you do 

```js
const newPool = new pg.Pool(otherPool.options)
console.log(JSON.stringify(newPool))
```
the password shows up again? Dang...yeah I mean ideally there's no way to __accidentally__ log out a password...regardless of how objects are constructed.

**Comment by charmander** [comment]
@brianc The password doesn’t show up, but that pattern will stop working, because the `Object.assign` that copies `otherPool.options` doesn’t include the `password` property.

**Comment by brianc** [comment]
oh i see what you're saying - so we could special case it to be copied in if you do that?  ugg....hmmm I mean we could special case copying password over, mutate the `options` argument and append the `Client` directly to it, make the `pg.Pool` constructor take a second (undocumented) argument for the Client constructor to use (gross).  We could also do something like...

```
class BoundPool {
 constructor (options) { 
    super(options)
    this.options.Client = Client
  }
}
```

That's actually pretty clean IMO and isn't even more code or anything.  Should fix the problem too, right?

**Comment by charmander** [comment]
@brianc Ah, nice solution! That would be perfect, except there’s this in pg-pool:

https://github.com/brianc/node-postgres/blob/ee8d32f97cd5e5907c8cd9d815c7fe57a7031f7f/packages/pg-pool/index.js#L65

which makes it do `require('pg')`, which is probably fine because of the peer dependency but still a little weird. However, I think `|| Client` might achieve exactly what’s needed:

```js
super(options, Client)
```

which also defaults to the `Client` being passed into `BoundPool` if one exists, instead of overwriting it. \o/

**Pull request merged**